packages/clay-progress-bar/src/ClayProgressBar.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import Icon from '@clayui/icon';
+import warning from 'warning';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Flag to indicate if `children` should be wrapped as `feedback`.
+	 */
+	feedback?: boolean;
+
+	/**
+	 * The current value of the progress bar. Should range from 0 to 100.
+	 */
+	value: number;
+
+	/**
+	 * Path to spritemap for icon symbol.
+	 */
+	spritemap?: string;
+
+	/**
+	 * Flag to indicate whether a "warning" color for the bar.
+	 */
+	warn?: boolean;
+}
+
+const ClayProgressBar: React.FunctionComponent<Props> = ({
+	children,
+	className,
+	feedback = false,
+	spritemap,
+	value = 0,
+	warn,
+	...otherProps
+}) => {
+	warning(
+		value >= 0 && value <= 100,
+		`ClayProgressBar requires \`value\` to be in the range of 0 to 100, it received ${value}.`
+	);
+
+	value = Math.min(Math.max(value, 0), 100);
```

Comment:
This is just a micro-optimization 🙂 But we can use `useMemo`.

```suggestion
	value = useMemo(() => Math.min(Math.max(value, 0), 100), [value]);
```

Reply:
It's probably a dangerous habit to get into to apply microoptimizations "just because". In this case, the `useMemo` version is harder to read, and it seems unlikely that it would deliver any measurable performance benefit.

See [this benchmark](https://jsperf.com/clamp-micro/1), for example:

![Math_min___Math_max_·_jsPerf](https://user-images.githubusercontent.com/7074/55816542-be221180-5af2-11e9-83ab-c64d1758bde2.png)

Clearly, it's so fast that the difference is just statistical noise.

Reply:
hey @wincent, Thanks for bringing this test, although it is relative and can vary... I agree that even if the use of `useMemo` comes to bring some benefit it will be something minimal or symbolic... (in this case)